src/beacon-api/app/websockets/routing.py

```
@@ -0,0 +1,299 @@
+def decode_jwt_user_id(token: str) -> Optional[str]:
+    """Helper to decode JWT payload safely and retrieve user _id field."""
+    try:
+        parts = token.split(".")
+        if len(parts) != 3:
+            return None
+        payload_b64 = parts[1]
+        payload_b64 += "=" * ((4 - len(payload_b64) % 4) % 4)
+        payload_bytes = base64.urlsafe_b64decode(payload_b64)
+        payload = json.loads(payload_bytes.decode("utf-8"))
+        return payload.get("_id")
+    except Exception as e:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)